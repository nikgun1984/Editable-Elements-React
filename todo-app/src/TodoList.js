import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { useState } from "react";
import { v4 as uuid } from "uuid";

const TodoList = () => {
	const [todos, setTodos] = useState([]);
	const addTodo = (toDo) => {
		const newTodo = { ...toDo, id: uuid() };
		setTodos((todos) => [...todos, newTodo]);
	};
	const removeTodo = (id) => {
		setTodos(todos.filter((todo) => todo.id !== id));
	};
	return (
		<div>
			<h1>Todo List</h1>
			<NewTodoForm addTodo={addTodo} />
			<ul>
				{todos.map((toDo) => (
					<Todo
						key={toDo.id}
						id={toDo.id}
						todo={toDo.todo}
						removeTodo={removeTodo}
					/>
				))}
			</ul>
		</div>
	);
};

export default TodoList;
