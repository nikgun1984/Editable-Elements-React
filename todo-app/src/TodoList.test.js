import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// smoke test
it("renders without crashing", function () {
	render(<TodoList />);
});

// snapshot test
it("matches snapshot", function () {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

// specialized testing
it("can add a new todo", function () {
	const { getByLabelText, queryByText } = render(<TodoList />);

	//no todos just yet
	expect(queryByText("Go Shopping")).not.toBeInTheDocument();

	const toDoItem = getByLabelText("Todo:");
	const btn = queryByText("Submit");

	// fill out the form
	fireEvent.change(toDoItem, { target: { value: "Go Shopping" } });
	fireEvent.click(btn);

	expect(queryByText("Go Shopping", { exact: false })).toBeInTheDocument();
});
