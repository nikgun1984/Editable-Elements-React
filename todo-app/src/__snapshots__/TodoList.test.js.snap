// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todo List
    </h1>
    <div>
      <form>
        <label
          for="todo"
        >
          Todo:
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="Enter Todo..."
          type="text"
          value=""
        />
        <button>
          Submit
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
